![example](images/director_shot.jpeg)

# Project Title

**Authors:** Jordan, Alex, Warren
***

## Overview

A one-paragraph overview of the project, including the business problem, data, methods, results and recommendations.

## Business Problem

Summary of the business problem you are trying to solve, and the data questions that you plan to answer to solve them.

***
Questions to consider:
* What are the business's pain points related to this project?
* How did you pick the data analysis question(s) that you did?
* Why are these questions important from a business perspective?
***

## Data Understanding

Describe the data being used for this project.
***
Questions to consider:
* Where did the data come from, and how do they relate to the data analysis questions?
* What do the data represent? Who is in the sample and what variables are included?
* What is the target variable?
* What are the properties of the variables you intend to use?
***

### Look at available data

First, I am looking at the information for each table. I am looking for the number of entries, column names, data types, and how many null values there are. I will not be using the bom.movie_gross data because I am not interested in the studio and the rest of the data is in tn.movies_budgets.

The datasets that I am using are:
    
    - name.basics
    - title.basics
    - title.akas
    - title.crew
    - title.ratings
    - title.principals

What does each dataset contain?
    
    - name.basics
        - nconst: a name id used for merging to other dataframes
        - primary_name: The name associated with each primary profession
        - birth_year: Birth year of primary_name
        - death_year: Death year of primary_name
        - primary_profession: comma separated professions related to the primary_name
        - known_titles: id numbers that represent movies that the primary_name had a part in
        
    - title.basics
        - tconst: a title id used for merging other dataframes
        - primary_title: the movie name
        - original_title: also the movie name
        - start_year: year that the movie was released
        - runtime_minutes: how long the movie is in minutes
        - genres: all of the genres associated with the movie

    These are merged into a dataframe called tcombo
    
    We will be using profit as a metric that we have defined to be $tcombo['worldwide_gross'] - tcombo['production_budget']$
    
    I will be looking at writers vs profit, region vs profit, and release month vs profit
    

In [1]:
# Import standard packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()
%matplotlib inline

#### Looking at the tables

In [2]:
nb = pd.read_csv('data/zippedData/imdb.name.basics.csv.gz')
nb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 606648 entries, 0 to 606647
Data columns (total 6 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   nconst              606648 non-null  object 
 1   primary_name        606648 non-null  object 
 2   birth_year          82736 non-null   float64
 3   death_year          6783 non-null    float64
 4   primary_profession  555308 non-null  object 
 5   known_for_titles    576444 non-null  object 
dtypes: float64(2), object(4)
memory usage: 27.8+ MB


In [3]:
#nb.head(1)
#actor names, years, profession, titles
#seems less than ideal
#might be able to join known_for_titles with tkas etc

In [4]:
takas = pd.read_csv('data/zippedData/imdb.title.akas.csv.gz')
takas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 331703 entries, 0 to 331702
Data columns (total 8 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   title_id           331703 non-null  object 
 1   ordering           331703 non-null  int64  
 2   title              331703 non-null  object 
 3   region             278410 non-null  object 
 4   language           41715 non-null   object 
 5   types              168447 non-null  object 
 6   attributes         14925 non-null   object 
 7   is_original_title  331678 non-null  float64
dtypes: float64(1), int64(1), object(6)
memory usage: 20.2+ MB


In [5]:
#takas.head(1)

In [6]:
tbase = pd.read_csv('data/zippedData/imdb.title.basics.csv.gz')
#tbase.columns

In [7]:
#tbase.head(1)
#might try to combine with above on title_id and tconst?

In [8]:
tcrew = pd.read_csv('data/zippedData/imdb.title.crew.csv.gz')
tcrew.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 3 columns):
 #   Column     Non-Null Count   Dtype 
---  ------     --------------   ----- 
 0   tconst     146144 non-null  object
 1   directors  140417 non-null  object
 2   writers    110261 non-null  object
dtypes: object(3)
memory usage: 3.3+ MB


In [9]:
#tcrew.head(1)

In [10]:
tprinc = pd.read_csv('data/zippedData/imdb.title.principals.csv.gz')
# tprinc.info()

In [11]:
#tprinc.head(1)
#unsure of what this data is, but looks like tconst can be used to combine

In [12]:
# trat = pd.read_csv('data/zippedData/imdb.title.ratings.csv.gz')
# trat.info()

In [13]:
#trat.head(1)
#combine this rating with tcrew, tbase, takas on tconst column

In [14]:
#tmdb = pd.read_csv('data/tmdb.movies.csv')
#tmdb.info()

#not using tmdb

In [15]:
#tmdb.head(1)

In [16]:
tn = pd.read_csv('data/zippedData/tn.movie_budgets.csv.gz')
tn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


Skip to Filter Files
>
/
Name
Last Modified

example
Project Title

Authors: Jordan, Alex, Warren
Overview

A one-paragraph overview of the project, including the business problem, data, methods, results and recommendations.
Business Problem

Summary of the business problem you are trying to solve, and the data questions that you plan to answer to solve them.

Questions to consider:

    What are the business's pain points related to this project?
    How did you pick the data analysis question(s) that you did?
    Why are these questions important from a business perspective?

## Data Understanding

​

Describe the data being used for this project.

***

Questions to consider:

* Where did the data come from, and how do they relate to the data analysis questions?

* What do the data represent? Who is in the sample and what variables are included?

* What is the target variable?

* What are the properties of the variables you intend to use?

***

​

### Look at available data

​

First, I am looking at the information for each table. I am looking for the number of entries, column names, data types, and how many null values there are. I will not be using the bom.movie_gross data because I am not interested in the studio and the rest of the data is in tn.movies_budgets.

​

The datasets that I am using are:

    

    - name.basics

    - title.basics

    - title.akas

    - title.crew

    - title.ratings

    - title.principals

​

What does each dataset contain?

    

    - name.basics

        - nconst: a name id used for merging to other dataframes

        - primary_name: The name associated with each primary profession

        - birth_year: Birth year of primary_name

        - death_year: Death year of primary_name

        - primary_profession: comma separated professions related to the primary_name

        - known_titles: id numbers that represent movies that the primary_name had a part in

        

    - title.basics

        - tconst: a title id used for merging other dataframes

        - primary_title: the movie name

        - original_title: also the movie name

        - start_year: year that the movie was released

        - runtime_minutes: how long the movie is in minutes

        - genres: all of the genres associated with the movie

​

    These are merged into a dataframe called tcombo

    

    We will be using profit as a metric that we have defined to be $tcombo['worldwide_gross'] - tcombo['production_budget']$

    

    I will be looking at writers vs profit, region vs profit, and release month vs profit

    

# Import standard packages

import pandas as pd

import numpy as np

import matplotlib.pyplot as plt

import seaborn as sns

sns.set()

%matplotlib inline

Looking at the tables

nb = pd.read_csv('data/zippedData/imdb.name.basics.csv.gz')

#nb.info()

#nb.head(1)

#actor names, years, profession, titles

#seems less than ideal

#might be able to join known_for_titles with tkas etc

takas = pd.read_csv('data/zippedData/imdb.title.akas.csv.gz')

#takas.info()

#takas.head(1)

tbase = pd.read_csv('data/zippedData/imdb.title.basics.csv.gz')

#tbase.columns

#tbase.head(1)

#might try to combine with above on title_id and tconst?

tcrew = pd.read_csv('data/zippedData/imdb.title.crew.csv.gz')

tcrew.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 3 columns):
 #   Column     Non-Null Count   Dtype 
---  ------     --------------   ----- 
 0   tconst     146144 non-null  object
 1   directors  140417 non-null  object
 2   writers    110261 non-null  object
dtypes: object(3)
memory usage: 3.3+ MB

#tcrew.head(1)

# tprinc = pd.read_csv('data/zippedData/imdb.title.principals.csv.gz')

# tprinc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1028186 entries, 0 to 1028185
Data columns (total 6 columns):
 #   Column      Non-Null Count    Dtype 
---  ------      --------------    ----- 
 0   tconst      1028186 non-null  object
 1   ordering    1028186 non-null  int64 
 2   nconst      1028186 non-null  object
 3   category    1028186 non-null  object
 4   job         177684 non-null   object
 5   characters  393360 non-null   object
dtypes: int64(1), object(5)
memory usage: 47.1+ MB

#tprinc.head(1)

#unsure of what this data is, but looks like tconst can be used to combine

# trat = pd.read_csv('data/zippedData/imdb.title.ratings.csv.gz')

# trat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   tconst         73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB

#trat.head(1)

#combine this rating with tcrew, tbase, takas on tconst column

#tmdb = pd.read_csv('data/tmdb.movies.csv')

#tmdb.info()

​

#not using tmdb

#tmdb.head(1)

tn = pd.read_csv('data/zippedData/tn.movie_budgets.csv.gz')

tn.info()

In [17]:
#tn.head(10)
#looks similar to bmg, but has movie name

In [18]:
#nb = pd.read_csv('data/name.basics.csv')
#nb.head()

## Data Preparation

Describe and justify the process for preparing the data for analysis.

***
Questions to consider:
* Were there variables you dropped or created?
* How did you address missing values or outliers?
* Why are these choices appropriate given the data and the business problem?
***

Now that I have looked at all of the information available, I am going to combine some of them into two dataframes.
The first, will be a data frame that contains the title csv's. These will be outer joined on the tconst column so that I don't loose any data. The tconst column is an id for each movie.

Next, I drop rows with null in all columns except the domestic and worldwide gross. It is fine if either domestic_gross or worldwide_gross is zero, but not both. I did not want to drop too much data so this is where I drew the line.

Our target variable is the profit that each movie makes. I added a profit column that is the difference in the worldwide gross and the budget of the film.


The first step is to load in all of the tables that I am using and them merge them together. I used the tconst column and nconst column to merge them. I used an outer merge for all of these so that I don't loose data in this step. Later, I will drop some rows with null values and drop some columns.

I will drop thses columns:
-    'language',
-    'types',
-    'attributes',
-    'is_original_title',
-    'ordering_x',
-    'ordering_y',
-    'original_title',
-    'numvotes',
-    'birth_year',
-    'death_year'

I am still left with a lot of null values so I will drop nulls from the following columns:
-    'tconst',
-    'averagerating',
-    'directors',
-    'writers',
-    'movie',
-    'start_year',
-    'runtime_minutes',
-    'genres',
-    'title',
-    'region',
-    'id',
-    'release_date',
-    'production_budget'

This is where I make sure that the tconst column name is consistient. I had to rename the column in title.akas before merging. I used an outer merge so that I can explicitly drop the data that I don't want.

In [19]:
#rename takas['title_id'] to takas['tconst'] to that we can use it to merge on
takas.rename(columns={'title_id':'tconst'},inplace=True)

In [20]:
#trat join with tcrew, tbase, takas on tconst column
#used outer to keep all data so I can drop later
tcombo = tcrew.merge(tbase,how='outer',on='tconst')
tcombo = tcombo.merge(takas,how='outer',on='tconst')
tcombo = tcombo.merge(tcrew,how='outer',on='tconst')
tcombo = tcombo.merge(tprinc,how='outer',on='tconst')
tcombo = tcombo.merge(nb,how='outer',on='nconst')
tcombo.head(1)

tconst directors_x  writers_x   primary_title  original_title  \
0  tt0285252   nm0899854  nm0899854  Life's a Beach  Life's a Beach   

   start_year  runtime_minutes  genres  ordering_x         title  ...  \
0      2012.0            100.0  Comedy         1.0  Bikini Beach  ...   

  ordering_y     nconst  category  job  characters  primary_name birth_year  \
0       10.0  nm1077681  composer  NaN         NaN  Fuad Javadov        NaN   

   death_year primary_profession                         known_for_titles  
0         NaN     actor,composer  tt0300480,tt0285252,tt1051834,tt0991289  

[1 rows x 27 columns]

It looks like all the columns are here, but the primary_title column needs to be renamed so we can use it to merge with tn.movie_budgets.

In [21]:
#change tcombo['primary_title'] to movie

tcombo.rename(columns={'primary_title':'movie'},inplace=True)
tcombo.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2978143 entries, 0 to 2978142
Data columns (total 27 columns):
 #   Column              Dtype  
---  ------              -----  
 0   tconst              object 
 1   directors_x         object 
 2   writers_x           object 
 3   movie               object 
 4   original_title      object 
 5   start_year          float64
 6   runtime_minutes     float64
 7   genres              object 
 8   ordering_x          float64
 9   title               object 
 10  region              object 
 11  language            object 
 12  types               object 
 13  attributes          object 
 14  is_original_title   float64
 15  directors_y         object 
 16  writers_y           object 
 17  ordering_y          float64
 18  nconst              object 
 19  category            object 
 20  job                 object 
 21  characters          object 
 22  primary_name        object 
 23  birth_year          float64
 24  death_year          floa

In [22]:
tcombo = tcombo.merge(tn,how='outer',on='movie')
#tcombo

Look at the columns now. primary_title was changed to movie

In [23]:
tcombo.head(1)

tconst directors_x  writers_x           movie  original_title  \
0  tt0285252   nm0899854  nm0899854  Life's a Beach  Life's a Beach   

   start_year  runtime_minutes  genres  ordering_x         title  ...  \
0      2012.0            100.0  Comedy         1.0  Bikini Beach  ...   

   primary_name birth_year death_year primary_profession  \
0  Fuad Javadov        NaN        NaN     actor,composer   

                          known_for_titles  id release_date  \
0  tt0300480,tt0285252,tt1051834,tt0991289 NaN          NaN   

   production_budget domestic_gross worldwide_gross  
0                NaN            NaN             NaN  

[1 rows x 32 columns]

Now its time to drop columns that aren't needed.

In [24]:
#next, look at which rows/columns need to be dropped bc nulls

#drop columns:
tcombo.drop(['language','types','attributes','is_original_title','ordering_x','ordering_y','original_title','birth_year','death_year'],axis=1,inplace=True)
#tcombo.info()

I have consolidated the data into a dataframe called tcombo.

After looking at how many null values there are in some of the columns, I am going to see what happens if I get rid of all the rows with any null values from a subset of the columns.

In [25]:
#I am going to not drop null gross rows

tcombo.dropna(subset=['tconst','averagerating','directors','writers','movie','start_year','runtime_minutes','genres','title','region','id','release_date','production_budget'],inplace=True)
tcombo.head(5)

KeyError: ['averagerating', 'directors', 'writers']

I noticed that there are some duplicates and I will get rid of them in the nect cell.

In [ ]:
#I had lots of duplicates so I dropped the dupes in tconst and movie
tcombo.drop_duplicates(subset=['tconst','movie'],inplace=True)
#tcombo.head(5)

Next, I change the data types of the budget and gross columns to be floats. I got rid of the dollar signs and commas and set them to type float.

In [ ]:
#change budget and gross types so int
tcombo['production_budget'] = tcombo['production_budget'].map(lambda x: x.replace('$',""))
tcombo['production_budget'] = tcombo['production_budget'].map(lambda x: x.replace(',',""))
tcombo['domestic_gross'] = tcombo['domestic_gross'].map(lambda x: x.replace('$',""))
tcombo['domestic_gross'] = tcombo['domestic_gross'].map(lambda x: x.replace(',',""))
tcombo['worldwide_gross'] = tcombo['worldwide_gross'].map(lambda x: x.replace('$',""))
tcombo['worldwide_gross'] = tcombo['worldwide_gross'].map(lambda x: x.replace(',',""))



I also made a column called release month that I got from the date column.

In [ ]:
#making a new column called release month
tcombo['release_month'] = tcombo['release_date'].map(lambda x: x[:3])
#tcombo

In [ ]:
#change the types to float
tcombo.production_budget = tcombo.production_budget.astype(float)
tcombo.domestic_gross = tcombo.domestic_gross.astype(float)
tcombo.worldwide_gross = tcombo.worldwide_gross.astype(float)


Check my column types

In [ ]:
tcombo.dtypes

Now it's time to make a profit column

In [ ]:
#make profit column
tcombo['profit'] = (tcombo['worldwide_gross'] - tcombo['production_budget'])
#tcombo.head(1)

This is where I pull out the rows where the production budget is not zero and either the domestic or worldwide gross are not zero. This is supposed to prevent too many profit rows from being negative. Otherwise, it looks like a lot of movies lose money.

In [ ]:
#drop rows in gross and budget that are zero
tcombo = tcombo.loc[((tcombo['domestic_gross'] != 0) | (tcombo['worldwide_gross'] != 0)) & (tcombo['production_budget'] != 0)]


#what is total US region profit
#protot_US = tcombo['profit'].loc[tcombo['region'] == 'US'].sum()
#protot_US

## Data Modeling
Describe and justify the process for analyzing or modeling the data.

***
Questions to consider:
* How did you analyze or model the data?
* How did you iterate on your initial approach to make it better?
* Why are these choices appropriate given the data and the business problem?
***


I would like to visualize some of the data now with some histograms. The goal is to figure out the best way to spent money. So what gives the best results?

- rating for each genre, director, region?
- what is the connection between rating and profit?
- look at total profit for each director, writers, genre, region, release month
- profitable genres for each month

### genre vs rating - Jordan did this one

In [ ]:
#tcombo_genre_rating = tcombo[['averagerating','genres']]
#tcombo_genre_rating
#split on ','
#tcombo_genre_rating[:]['genres'] = tcombo_genre_rating['genres'].str.split(',')
#tcombo_genre_rating

In [ ]:
#tcombo_genre_profit = tcombo[['genres','profit']]
#tcombo_genre_profit[:]['genres'] = tcombo_genre_profit['genres'].str.split(',')


In [ ]:
#np.unique(tcombo_genre_rating['genres'].sum())


In [ ]:
#tcombo_genre_rating.genres.value_counts()
#column of lists so its not working
#use .explode
#tcombo_genre_profit = tcombo_genre_profit.explode('genres')
#profit_group = tcombo_genre_profit.groupby(['genres']).sum().sort_values(by='profit',ascending=False)[:10]
#profit_group

In [ ]:
#sns.set(rc={'figure.figsize':(15,8)})
#sns.set_theme(context='notebook',style='darkgrid')

#ax = sns.barplot(x=profit_group.index,y='profit',data=profit_group,color='cornflowerblue')

#seems the right order of magnitude so thats a good sign

In [ ]:

# tcombo_genre_rating = tcombo_genre_rating.sort_values(by='averagerating',ascending=False)[:11]

# sns.set(rc={'figure.figsize':(15,8)})   

# sns.barplot('genres','averagerating',data=tcombo_genre_rating)

Looks a little like a better rating gives higher profit

In [ ]:
#rating and profit
#tcombo_rating_profit = tcombo[['averagerating','profit']]

#fig, ax = plt.subplots(figsize=(15,8))

#x = tcombo_rating_profit['averagerating']
#y = tcombo_rating_profit['profit']

#ax.scatter(x,y)

### director vs profit - I think Jordan did this one too

Here, I'm making a barplot to compare the most profitable directors.

First, I make a new dataframe that contains the director id, profit, primary_name, and primary_profession. The director and primary_profession columns contain string objects with are multiple ids and professions separated by commas. I use .split to turn the data into a list of strings. Then, I call .explode on each list so that there is only one value per row. 

To plot just directors or just writers, I found the rows where the primary_profession was director or writer and made a new dataframe with those values and profit to make the graph with.

In [ ]:
tcombo_profession_profit = tcombo[['directors','profit','primary_name','primary_profession']]
tcombo_profession_profit[:]['directors'] = tcombo_profession_profit['directors'].str.split(',')
tcombo_profession_profit[:]['primary_profession'] = tcombo_profession_profit['primary_profession'].str.split(',')
tcombo_profession_profit.head(3)

Next, turn the column of lists into multiple rows (one row for each entry in the list)

I also replace the spaces in the primary name column with new lines so that the labels on the barplots look better.

In [ ]:
#use .explode

tcombo_profession_profit = tcombo_profession_profit.explode('directors')

#replace spaces with new line so that the names look better on the graphs
tcombo_profession_profit.primary_name = tcombo_profession_profit.primary_name.map(lambda x: x.replace(' ','\n'))


tcombo_profession_profit = tcombo_profession_profit.explode('primary_profession')
tcombo_profession_profit.head(3)

Find all rows where the profession column is director

In [ ]:
#find all rows where the primary profession is director
director_group = tcombo_profession_profit.loc[tcombo_profession_profit['primary_profession'] == 'director']
director_group = director_group.groupby(['primary_name']).sum().sort_values(by='profit',ascending=False)[:10]
director_group.head(3)

Plot top profitable directors

In [ ]:
#director and profit
# director_profit


sns.set(rc={'figure.figsize':(15,8)})   

ax = sns.barplot(x=director_group.index,y='profit',data=director_group,color='cornflowerblue')
#these are the directors who bring in the most money

ax.set(xlabel='Director',ylabel='Profit - Billions of Dollars',title='Most Profitable Directors');

### profit vs writer
The next plot was done that same way as above. I isolated the rows where the profession was 'writer' and plotted those in the most profitable movies.

In [ ]:
writer_group = tcombo_profession_profit.loc[tcombo_profession_profit['primary_profession'] == 'writer']

writer_group = writer_group.groupby(['primary_name']).sum().sort_values(by='profit',ascending=False)[:10]
writer_group.head(3)

In [ ]:

sns.set(rc={'figure.figsize':(15,8)})   

ax = sns.barplot(x=writer_group.index,y='profit',data=writer_group,color='cornflowerblue')

ax.set(xlabel='Writer',ylabel='Profit - Billions of Dollars',title='Most Profitable Writers');

### profit vs region

This is a bar plot of the regions where movies made the most money. I found a dictionary of country codes and their country names. I used it to map the country names in place of the codes in the region column. This makes the plot labels more meaningful.

#### Country Code Dictionary

I found a dictionary of country codes so I can map them and use the actual country name on the graph.

In [ ]:
CC = {
    "AF": "AFGHANISTAN",
    "AX": "ÅLAND ISLANDS",
    "AL": "ALBANIA",
    "DZ": "ALGERIA",
    "AS": "AMERICAN SAMOA",
    "AD": "ANDORRA",
    "AO": "ANGOLA",
    "AI": "ANGUILLA",
    "AQ": "ANTARCTICA",
    "AG": "ANTIGUA AND BARBUDA",
    "AR": "ARGENTINA",
    "AM": "ARMENIA",
    "AW": "ARUBA",
    "AU": "AUSTRALIA",
    "AT": "AUSTRIA",
    "AZ": "AZERBAIJAN",
    "BS": "BAHAMAS",
    "BH": "BAHRAIN",
    "BD": "BANGLADESH",
    "BB": "BARBADOS",
    "BY": "BELARUS",
    "BE": "BELGIUM",
    "BZ": "BELIZE",
    "BJ": "BENIN",
    "BM": "BERMUDA",
    "BT": "BHUTAN",
    "BO": "BOLIVIA, PLURINATIONAL STATE OF",
    "BQ": "BONAIRE, SINT EUSTATIUS AND SABA",
    "BA": "BOSNIA AND HERZEGOVINA",
    "BW": "BOTSWANA",
    "BV": "BOUVET ISLAND",
    "BR": "BRAZIL",
    "IO": "BRITISH INDIAN OCEAN TERRITORY",
    "BN": "BRUNEI DARUSSALAM",
    "BG": "BULGARIA",
    "BF": "BURKINA FASO",
    "BI": "BURUNDI",
    "KH": "CAMBODIA",
    "CM": "CAMEROON",
    "CA": "CANADA",
    "CV": "CAPE VERDE",
    "KY": "CAYMAN ISLANDS",
    "CF": "CENTRAL AFRICAN REPUBLIC",
    "TD": "CHAD",
    "CL": "CHILE",
    "CN": "CHINA",
    "CX": "CHRISTMAS ISLAND",
    "CC": "COCOS (KEELING) ISLANDS",
    "CO": "COLOMBIA",
    "KM": "COMOROS",
    "CG": "CONGO",
    "CD": "CONGO, THE DEMOCRATIC REPUBLIC OF THE",
    "CK": "COOK ISLANDS",
    "CR": "COSTA RICA",
    "CI": "CÔTE D'IVOIRE",
    "HR": "CROATIA",
    "CU": "CUBA",
    "CW": "CURAÇAO",
    "CY": "CYPRUS",
    "CZ": "CZECH REPUBLIC",
    "DK": "DENMARK",
    "DJ": "DJIBOUTI",
    "DM": "DOMINICA",
    "DO": "DOMINICAN REPUBLIC",
    "EC": "ECUADOR",
    "EG": "EGYPT",
    "SV": "EL SALVADOR",
    "GQ": "EQUATORIAL GUINEA",
    "ER": "ERITREA",
    "EE": "ESTONIA",
    "ET": "ETHIOPIA",
    "FK": "FALKLAND ISLANDS (MALVINAS)",
    "FO": "FAROE ISLANDS",
    "FJ": "FIJI",
    "FI": "FINLAND",
    "FR": "FRANCE",
    "GF": "FRENCH GUIANA",
    "PF": "FRENCH POLYNESIA",
    "TF": "FRENCH SOUTHERN TERRITORIES",
    "GA": "GABON",
    "GM": "GAMBIA",
    "GE": "GEORGIA",
    "DE": "GERMANY",
    "GH": "GHANA",
    "GI": "GIBRALTAR",
    "GR": "GREECE",
    "GL": "GREENLAND",
    "GD": "GRENADA",
    "GP": "GUADELOUPE",
    "GU": "GUAM",
    "GT": "GUATEMALA",
    "GG": "GUERNSEY",
    "GN": "GUINEA",
    "GW": "GUINEA-BISSAU",
    "GY": "GUYANA",
    "HT": "HAITI",
    "HM": "HEARD ISLAND AND MCDONALD ISLANDS",
    "VA": "HOLY SEE (VATICAN CITY STATE)",
    "HN": "HONDURAS",
    "HK": "HONG KONG",
    "HU": "HUNGARY",
    "IS": "ICELAND",
    "IN": "INDIA",
    "ID": "INDONESIA",
    "IR": "IRAN, ISLAMIC REPUBLIC OF",
    "IQ": "IRAQ",
    "IE": "IRELAND",
    "IM": "ISLE OF MAN",
    "IL": "ISRAEL",
    "IT": "ITALY",
    "JM": "JAMAICA",
    "JP": "JAPAN",
    "JE": "JERSEY",
    "JO": "JORDAN",
    "KZ": "KAZAKHSTAN",
    "KE": "KENYA",
    "KI": "KIRIBATI",
    "KP": "KOREA, DEMOCRATIC PEOPLE'S REPUBLIC OF",
    "KR": "KOREA, REPUBLIC OF",
    "KW": "KUWAIT",
    "KG": "KYRGYZSTAN",
    "LA": "LAO PEOPLE'S DEMOCRATIC REPUBLIC",
    "LV": "LATVIA",
    "LB": "LEBANON",
    "LS": "LESOTHO",
    "LR": "LIBERIA",
    "LY": "LIBYA",
    "LI": "LIECHTENSTEIN",
    "LT": "LITHUANIA",
    "LU": "LUXEMBOURG",
    "MO": "MACAO",
    "MK": "MACEDONIA, THE FORMER YUGOSLAV REPUBLIC OF",
    "MG": "MADAGASCAR",
    "MW": "MALAWI",
    "MY": "MALAYSIA",
    "MV": "MALDIVES",
    "ML": "MALI",
    "MT": "MALTA",
    "MH": "MARSHALL ISLANDS",
    "MQ": "MARTINIQUE",
    "MR": "MAURITANIA",
    "MU": "MAURITIUS",
    "YT": "MAYOTTE",
    "MX": "MEXICO",
    "FM": "MICRONESIA, FEDERATED STATES OF",
    "MD": "MOLDOVA, REPUBLIC OF",
    "MC": "MONACO",
    "MN": "MONGOLIA",
    "ME": "MONTENEGRO",
    "MS": "MONTSERRAT",
    "MA": "MOROCCO",
    "MZ": "MOZAMBIQUE",
    "MM": "MYANMAR",
    "NA": "NAMIBIA",
    "NR": "NAURU",
    "NP": "NEPAL",
    "NL": "NETHERLANDS",
    "NC": "NEW CALEDONIA",
    "NZ": "NEW ZEALAND",
    "NI": "NICARAGUA",
    "NE": "NIGER",
    "NG": "NIGERIA",
    "NU": "NIUE",
    "NF": "NORFOLK ISLAND",
    "MP": "NORTHERN MARIANA ISLANDS",
    "NO": "NORWAY",
    "OM": "OMAN",
    "PK": "PAKISTAN",
    "PW": "PALAU",
    "PS": "PALESTINE, STATE OF",
    "PA": "PANAMA",
    "PG": "PAPUA NEW GUINEA",
    "PY": "PARAGUAY",
    "PE": "PERU",
    "PH": "PHILIPPINES",
    "PN": "PITCAIRN",
    "PL": "POLAND",
    "PT": "PORTUGAL",
    "PR": "PUERTO RICO",
    "QA": "QATAR",
    "RE": "RÉUNION",
    "RO": "ROMANIA",
    "RU": "RUSSIAN FEDERATION",
    "RW": "RWANDA",
    "BL": "SAINT BARTHÉLEMY",
    "SH": "SAINT HELENA, ASCENSION AND TRISTAN DA CUNHA",
    "KN": "SAINT KITTS AND NEVIS",
    "LC": "SAINT LUCIA",
    "MF": "SAINT MARTIN (FRENCH PART)",
    "PM": "SAINT PIERRE AND MIQUELON",
    "VC": "SAINT VINCENT AND THE GRENADINES",
    "WS": "SAMOA",
    "SM": "SAN MARINO",
    "ST": "SAO TOME AND PRINCIPE",
    "SA": "SAUDI ARABIA",
    "SN": "SENEGAL",
    "RS": "SERBIA",
    "SC": "SEYCHELLES",
    "SL": "SIERRA LEONE",
    "SG": "SINGAPORE",
    "SX": "SINT MAARTEN (DUTCH PART)",
    "SK": "SLOVAKIA",
    "SI": "SLOVENIA",
    "SB": "SOLOMON ISLANDS",
    "SO": "SOMALIA",
    "ZA": "SOUTH AFRICA",
    "GS": "SOUTH GEORGIA AND THE SOUTH SANDWICH ISLANDS",
    "SS": "SOUTH SUDAN",
    "ES": "SPAIN",
    "LK": "SRI LANKA",
    "SD": "SUDAN",
    "SR": "SURINAME",
    "SJ": "SVALBARD AND JAN MAYEN",
    "SZ": "SWAZILAND",
    "SE": "SWEDEN",
    "CH": "SWITZERLAND",
    "SY": "SYRIAN ARAB REPUBLIC",
    "TW": "TAIWAN, PROVINCE OF CHINA",
    "TJ": "TAJIKISTAN",
    "TZ": "TANZANIA, UNITED REPUBLIC OF",
    "TH": "THAILAND",
    "TL": "TIMOR-LESTE",
    "TG": "TOGO",
    "TK": "TOKELAU",
    "TO": "TONGA",
    "TT": "TRINIDAD AND TOBAGO",
    "TN": "TUNISIA",
    "TR": "TURKEY",
    "TM": "TURKMENISTAN",
    "TC": "TURKS AND CAICOS ISLANDS",
    "TV": "TUVALU",
    "UG": "UGANDA",
    "UA": "UKRAINE",
    "AE": "UNITED ARAB EMIRATES",
    "GB": "UNITED KINGDOM",
    "US": "UNITED STATES",
    "UM": "UNITED STATES MINOR OUTLYING ISLANDS",
    "UY": "URUGUAY",
    "UZ": "UZBEKISTAN",
    "VU": "VANUATU",
    "VE": "VENEZUELA, BOLIVARIAN REPUBLIC OF",
    "VN": "VIET NAM",
    "VG": "VIRGIN ISLANDS, BRITISH",
    "VI": "VIRGIN ISLANDS, U.S.",
    "WF": "WALLIS AND FUTUNA",
    "EH": "WESTERN SAHARA",
    "YE": "YEMEN",
    "ZM": "ZAMBIA",
    "ZW": "ZIMBABWE",
}

### Region Plot

A barplot of the profitable regions (countries)

Below is where I do the actual mapping with a lambda function.

In [ ]:
tcombo_profit_region = tcombo[['profit','region']].sort_values(by='profit',ascending=False)[:12]

#map cCodes to tcombo_profit_region
tcombo_profit_region['region'] = tcombo_profit_region['region'].map(lambda x: CC[x].title())
tcombo_profit_region.head(3)

In [ ]:
sns.set_theme(context='notebook',style="darkgrid")

sns.set(rc={'figure.figsize':(15,8)})   

ax = sns.barplot(x='region',y='profit',data=tcombo_profit_region,color='cornflowerblue')

ax.set(xlabel='Region',ylabel='Profit - Billions of Dollars',title='Most Profitable Regions');

### Actors and actresses in the most profitable movies

This is a subplot with two plots. The first shows actors and the second shows actresses. I first got just the rows where the profession was either actor or actress and made them their own dataframes. Then, I grouped them by the name of the person and summed the profits of all the movies they were in. Next, I sorted the people by the profit and selected the top 10. After that, I just plotted the top actors and actresses in the most profitable movies.

In [ ]:
actor_group = tcombo_profession_profit.loc[tcombo_profession_profit['primary_profession'] == 'actor']
actor_group = actor_group.groupby(['primary_name']).sum().sort_values(by='profit',ascending=False)[:5]

actress_group = tcombo_profession_profit.loc[tcombo_profession_profit['primary_profession'] == 'actress']
actress_group = actress_group.groupby(['primary_name']).sum().sort_values(by='profit',ascending=False)[:5]


In [ ]:
fig, axes = plt.subplots(nrows=1,ncols=2,figsize=(20,10))
fig.suptitle('Top Actors and Actresses in the Most Profitable Movies')

sns.barplot(ax=axes[0],x=actor_group.index,y='profit',data=actor_group,color='cornflowerblue')
axes[0].set(xlabel='Actor',ylabel='Profit - Billions of Dollars',title='Actors')

sns.barplot(ax=axes[1],x=actress_group.index,y='profit',data=actress_group,color='cornflowerblue')
axes[1].set(xlabel='Actress',ylabel='Profit - Billions of Dollars',title='Actresses');


### Release month vs profit

The first plot shows the total profit made per release month. The second shows the total number of movies released per month. I suspected that the beginning of the summer would be where the most money was made and when most movies were released. It looks like June movies made the most money, December has the higher number of movies released.

I should look into the best genres the release per month.

#### 

In [ ]:
month_order = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
#sort months so they are in order
tcombo[:]['release_month'] = pd.Categorical(tcombo['release_month'], month_order)

tcombo = tcombo.sort_values(by='release_month',ascending=False)

sns.set(rc={'figure.figsize':(15,8)})   

ax = sns.barplot(x='release_month',y='profit',data=tcombo,color='cornflowerblue')
#I think this plot gives the average profits per month

ax.set(xlabel='Month of Release',ylabel='Profit - Billions of Dollars',title='Most Profitable Release Month')

In [ ]:
#months = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
#sort months so they are in order
tcombo['release_month'] = pd.Categorical(tcombo['release_month'], ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'])

tcombo = tcombo.sort_values(by='release_month',ascending=False)

sns.set(rc={'figure.figsize':(15,8)})   

ax = sns.countplot(x='release_month',data=tcombo,color='cornflowerblue')
#how many movies are released each month

ax.set(xlabel='Release Month',ylabel='Number of Movies Released',title='Movies Released per Month')

#### Best Movie Genres to Release per Month

First, I make a new dataframe with the genres, release months, and profits. The genres are in a comma separated string so I split them on the comma and then use .explode on the lists so that each genre is in its own column.

In [ ]:
release_genre = tcombo[['genres','release_month','profit']]
release_genre[:]['genres'] = release_genre['genres'].str.split(',')
release_genre = release_genre.explode('genres')

release_genre.dropna(axis=0,how='any',subset=['profit'],inplace=True)
release_genre.head(3)

I wrote a function to give a dataframe of profits by genre for any month passed to the function. I layed out my logic below.

In [ ]:
#laying out logic for function below

# jan_profit = release_genre.loc[release_genre['release_month'] == 'Jan']

# jan_profit = jan_profit.groupby(['release_month','genres']).sum()
# jan_profit.dropna(axis=0,how='any',subset=['profit'],inplace=True)

# jan_profit.sort_values(by='profit',ascending=False)

In [ ]:
#returns the profit of each genre sorted descending for the given month
def month_profit(month,release_genre):
    month_profit = release_genre.loc[release_genre['release_month'] == month]
    
    #removed this line so that it is not grouped by release month anymore
    #    month_profit = month_profit.groupby(['release_month','genres']).sum()
    
    month_profit = month_profit.groupby(['genres']).sum()
    month_profit.dropna(axis=0,how='any',subset=['profit'],inplace=True)
    
    month_profit = month_profit.sort_values(by='profit',ascending=False)
    
    return month_profit

I was testing the output of the function so that I could use it in a for loop.

In [ ]:
df=month_profit('Jan',release_genre)

# fig, axes = plt.subplots(nrows=1,ncols=2,figsize=(20,10))
# fig.suptitle('Top Actors and Actresses in the Most Profitable Movies')

# sns.barplot(ax=axes[0],x=actor_group.index,y='profit',data=actor_group,color='cornflowerblue')
# axes[0].set(xlabel='Actor',ylabel='Profit - Billions of Dollars',title='Actors')

# sns.barplot(ax=axes[1],x=actress_group.index,y='profit',data=actress_group,color='cornflowerblue')
# axes[1].set(xlabel='Actress',ylabel='Profit - Billions of Dollars',title='Actresses')
# mi = pd.MultiIndex.from_frame(df)
# mi.to_frame(index=True)

#df


This loop makes a separate plot of genre profits for each month. I iterated through a list that I had already made called month_order. I was using it before to define the order of months in a barplot.

In [ ]:
#trying to multi index to get 'genres' from df in the loop.



for month in month_order:
    
    fig, ax = plt.subplots(nrows=1,ncols=1,figsize=(12,6))

    
    df=month_profit(month,release_genre)
    df = df[:6]
    #print(df)
    #print(df.index)
    
    ax = sns.barplot(x=df.index,y='profit',data=df,color='cornflowerblue')
    ax.set(xlabel='Genre',ylabel='Profit - Billions of Dollars',title=f'Most Profitable Genres for {month.title()}')
    


It is interesting to see which genres are most profitable per month. For example: October-Thriller ; February-Romance is 5th

Maybe October is spooky month, February is romance month, and everything else is action/adventure.

## Evaluation
Evaluate how well your work solves the stated business problem.

***
Questions to consider:
* How do you interpret the results?
* How well does your model fit your data? How much better is this than your baseline model?
* How confident are you that your results would generalize beyond the data you have?
* How confident are you that this model would benefit the business if put into use?
***

## Conclusions
Provide your conclusions about the work you've done, including any limitations or next steps.

***
Questions to consider:
* What would you recommend the business do as a result of this work?
* What are some reasons why your analysis might not fully solve the business problem?
* What else could you do in the future to improve this project?
***